src/Glimpse.Agent.AspNet/Inspectors/Tab.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using Glimpse.Agent.AspNet.Messages;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Glimpse.Agent.Inspectors
+{
+    // TODO: Is this what we want to name this type?
+    public abstract class Tab : Inspector
+    {
+        private IAgentBroker _broker;
+        public IAgentBroker Broker
+        {
+            get
+            {
+                if (_broker == null)
+                {
+                    var httpContextAccessor = new HttpContextAccessor();
+                    _broker = httpContextAccessor.HttpContext.ApplicationServices.GetService<IAgentBroker>();
+                }
+
+                return _broker;
+            }
+            set { _broker = value; }
+        }
+
+        public override void Before(HttpContext context)
```

Comment:
Are we fine with the fact that these are still exposed to users? I'm wondering if we are better off just implementing `IInspector` instead, that way we could explicitly implement `Before`/`After` and it would be harder for users to muck up?